runtime/lua/vim/_defaults.lua

```diff
@@ -159,53 +164,73 @@ do
   ---
   --- See |grr|, |grn|, |gra|, |i_CTRL-S|.
   do
-    vim.keymap.set('n', 'grn', function()
+    vim.keymap.set('n', { desc = 'vim.lsp.buf.rename()' }, 'grn', function()
       vim.lsp.buf.rename()
-    end, { desc = 'vim.lsp.buf.rename()' })
+    end)
 
-    vim.keymap.set({ 'n', 'x' }, 'gra', function()
+    vim.keymap.set({ 'n', 'x' }, { desc = 'vim.lsp.buf.code_action()' }, 'gra', function()
       vim.lsp.buf.code_action()
-    end, { desc = 'vim.lsp.buf.code_action()' })
+    end)
 
-    vim.keymap.set('n', 'grr', function()
+    vim.keymap.set('n', { desc = 'vim.lsp.buf.references()' }, 'grr', function()
       vim.lsp.buf.references()
-    end, { desc = 'vim.lsp.buf.references()' })
+    end)
```

Comment:
This kind of case seems like a clear readability improvement.

Reply:
I personally think it's a sideways change. To me both are just as readble.

Reply:
Throwing in my unsolicited 2 cents, but I think it is a readability improvement for larger functions (and somewhat sideways change for 1 line like in this case).

Before:

```lua
vim.keymap.set('n', 'grr', function()
    local lang = vim.treesitter.language.inspect('c')
    local keys, symbols = {}, {}
    for k, v in pairs(lang) do
        if type(v) == 'boolean' then
            keys[k] = v
        else
            keys[k] = true
        end
    end
    return { keys, lang.fields, symbols }
end, { desc = 'vim.lsp.buf.references()' })
```

After:

```lua
vim.keymap.set('n', 'grr', { desc = 'vim.lsp.buf.references()' }, function()
    local lang = vim.treesitter.language.inspect('c')
    local keys, symbols = {}, {}
    for k, v in pairs(lang) do
        if type(v) == 'boolean' then
            keys[k] = v
        else
            keys[k] = true
        end
    end
    return { keys, lang.fields, symbols }
end)
```

Reply:
I think the litmus test is with bigger `opts` tables so stylua puts them on multiple lines. If the fit into a single line, it really makes no difference whether it is before the `function()` or after the `end`.

(Personally I -- much -- prefer the latter, since the `end)` line is wasted anyway, so might as well put something meaningful there. And having the `opts` before the `function` can push that too far to the right, making the arguments less visible.)

Reply:
@dundargoc yeah, that's definitely true. What discouraged me is that if the options list gets big, it's kind of "back to square one".

With a big `opts` list, the LHS is sitting the middle of a bunch of stuff:

```lua
vim.keymap.set(
  'n',
  { desc = 'vim.lsp.buf.references()', ... },
  'grr',
  function()
    local lang = vim.treesitter.language.inspect('c')
    local keys, symbols = {}, {}
    for k, v in pairs(lang) do
      if type(v) == 'boolean' then
        keys[k] = v
      else
        keys[k] = true
      end
    end
    return { keys, lang.fields, symbols }
  end)
```

And since there's little enthusiasm for it I'm just gonna punt. 

I'm happy to revive it again later if that changes; the logic changes were [small](https://github.com/neovim/neovim/commit/a96a0dbdddfcd9102d177c07355587a5890d6dec). 

Meanwhile, I started to doubt that the cost of "multiple ways to do something" was being offset.